doc: Add a flake.nix to ease build environment setup

ithinuel opened the pull request.

ithinuel committed:
doc: Add a flake.nix to ease build environment setup

9names commented on README.md:
```diff
@@ -55,6 +55,8 @@ If you aren't using a debugger (or want to use other debugging configurations),
 <details open="open">
   <summary><h2 style="display: inline-block" id="installation-of-development-dependencies">Installation of development dependencies</h2></summary>
 
+If you are using Nix, you can use the `devShell` made available in `flake.nix`. Otherwise, you can follow these commands (provided you already have [`Rustup`](https://rustup.rs/) installed):
```
I can use Nix without running NixOS, but this information and the flake are going to be of no use to me unless I'm running NixOS.
Some clarification of the target audience would be nice - also keeping in mind that a lot of the folks reading this won't know what Nix/NixOS is and if it's something we need them to install.

ithinuel replied:
> but this information and the flake are going to be of no use to me unless I'm running NixOS.

🤔 I’m using nix on MacOS and can use flakes there just fine. What would make NixOS required ?

9names replied:
Some sort of breadcrumb to help that a nix newbie like me could follow would be nice, yes. :)

9names replied:
so i kinda got it to work, but i'm not sure i did it right?
i installed nix, found i needed to add

```
extra-experimental-features = flakes nix-command
```

to `~/.config/nix/nix.conf`

and then i could run `nix develop --command [insert tool i wanted to run]`

like

```shell
$ nix develop --command cargo run
```

does that sound like how this is intended to be used?

ithinuel replied:
Yes, I added this command (well the build instead of run) to the GH workflow 👍

thejpster commented:
I had flake files in my Neotron repo but in the end I had to take them out because I wasn't testing them and they bit-rotted when I updated stuff.

ithinuel commented:
I see it regularly indeed. I've seen some gh actions that can be used to run flake checks & builds. I'll augment this pr when I get some time!

ithinuel committed:
feat: Add github workflow to validate the flake.nix and prevent bitrot

This shall hopefully help us catch breakage and future bitrot.
At least to let me know of issues so that I can help fixing.

Signed-off-by: Wilfried Chauveau <wilfried.chauveau@ithinuel.me>
